### PR TITLE
Update CPU pinning based on platform and streams detail

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *Update default CPU pinning based on platform and streams detail*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/inference/src/dev/threading/cpu_map_scheduling.cpp [hunk 1]
```diff
@@ -57,4 +58,33 @@ std::vector<std::vector<int>> apply_hyper_threading(bool input_value,
...
+                        const bool input_changed,
+                        const std::vector<std::vector<int>>& stream_info_table,
+                        const std::vector<std::vector<int>>& original_proc_type_table) {
+#if (defined(_WIN32) || defined(_WIN64))
+
+    return false;
+
+#endif
+
+#ifdef __linux__
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/threading/cpu_map_scheduling.cpp:71]: what's about other platforms? macOS? android?

File: src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp [hunk 2]
```diff
@@ -35,4 +35,16 @@ std::vector<std::vector<int>> apply_hyper_threading(bool input_type,
...
 
+/**
+ * @brief      Update CPU pinning property according to platform and user setting
+ * @param[in]  input_pinning indicates value of property enable_cpu_pinning.
+ * @param[in]  input_changed indicates if value is set by user.
+ * @param[in]  stream_info_table indicates the stream details that will be created..
+ * @param[in]  original_proc_type_table indicates CPU details on this platform
+ * @return     updated proc_type_table which removed unmatched processors
+ */
+bool update_cpu_pinning(const bool input_pinning,
```
Comment (ilya-lavrenov) [inline src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp:46]: as I see, this function is not used currently. Why do we need it then?
Secondly, it's not exported. So, it cannot be used outside of OpenVINO library. For example CPU plugin cannot use it, so does it really need to be in API folder?

Comment (wangleis) [reply src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp]: Move to CPU plugin folder as internal API.

File: src/inference/src/dev/threading/cpu_map_scheduling.cpp (hunk 1 above)
Comment (wangleis) [reply src/inference/src/dev/threading/cpu_map_scheduling.cpp]: This function is part of CPU map logic which is only enabled on Linux and Windows now.

When we enable CPU map for other platform, we will update this function as well.

File: src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp (hunk 2 above)
Comment (ilya-lavrenov) [reply src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp]: but can it be useful for GPU as well?
is it used in implementation of `enable_thread_pinning` ?

Comment (wangleis) [reply src/inference/dev_api/openvino/runtime/threading/cpu_map_scheduling.hpp]: This is the last step of new flow. GPU can use it as well. Move to `threading` folder.

**Pull request closed without merging**